assets/src/scripts/components/hydrograph/domain.js

```diff
@@ -112,31 +111,96 @@ export const getLowestAbsoluteValueOfTickValues = function(tickValues) {
 };
 
 /**
- * Function creates a new set of tick values that will fill in gaps in log scale ticks, then combines this new set with the
- * original set of tick marks.
- * @param {array} tickValues, the list of y-axis tick values
- * @returns {array} fullArrayOfTickMarks, the new full array of tick marks for log scales
+ * Helper function that generates roughly equally spaced tick values by taking the previous value and dividing it by 2
+ * to produce the next lower value
+ * @param {number}lowestTickValueOfLogScale, lowest numerical value of positive y-axis values, or lowest absolute
+ * value of negative y-axis values
+ * @returns {array} additionalTickValues, set of new y-axis tick values that will fill tick mark gaps on log scale graphs
  */
-export const getArrayOfAdditionalTickMarks = function(tickValues) {
+export const generateAdditionalTickValues = function(lowestTickValueOfLogScale) {
     let additionalTickValues = [];
-    let lowestTickValueOfLogScale = getLowestAbsoluteValueOfTickValues(tickValues);
-
-    // Make a set of tick values that when graphed will have equal spacing on the y axis
     while (lowestTickValueOfLogScale > 2) {
         lowestTickValueOfLogScale = Math.ceil(lowestTickValueOfLogScale / 2);
         additionalTickValues.push(lowestTickValueOfLogScale);
     }
+    return additionalTickValues;
+};
 
-    // if the log scale has negative values, add additional negative ticks with negative labels
+/**
+ * Helper function that rounds numerical values in an array based on the numerical value of the individual array item
+ * and a somewhat arbitrary numeric (rounding) value. The value in the array is then rounded to a multiple of the
+ * arbitrary numeric rounding value.
+ * @param {array} additionalTickValues, numerical values for y-axis ticks
+ * @param {array} yDomain, an array of two values, the lower and upper extent of the y-axis
+ * @returns {array} roundedTickValues, numerical values for y-axis ticks rounded to a multiple of a given number
+ */
+export const getRoundedTickValues = function(additionalTickValues, yDomain) {
+    let roundedTickValues = [];
+    // round the values based on an arbitrary breakpoints and rounding targets (may result in duplicate array values)
+    additionalTickValues.forEach(function(value) {
+        let roundingFactor = 1;
+        // first check that the value is greater than the position where the x-axis intersects the y-axis, then round
+        if (value > yDomain[0]) {
+            if (value > 10000) {
+               roundingFactor = 10000;
```

Comment:
I was wondering if maybe a more general solution might be possible, using Math.log10 / Math.pow. I think this is fine as-is, though.

Reply:
Yeah, I don't like the clunky look of the code here, but I did mess with a couple 'one-size-fits-all' attempts and I couldn't find anything that looked good across the gamut of data sets. :(